From c1554b6404351df0b3eacf9d551c9da16d38ba3b Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Tue, 19 Dec 2017 22:57:26 -0500
Subject: [PATCH] ExperienceOrbMergeEvent

Fired when the server is about to merge 2 experience orbs
Plugins can cancel this if they want to ensure experience orbs do not lose important
metadata such as spawn reason, or conditionally move data from source to target.

diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 8fca14243..8d3dca2a9 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -509,7 +509,7 @@ public class CraftEventFactory {
                     if (e instanceof EntityExperienceOrb) {
                         EntityExperienceOrb loopItem = (EntityExperienceOrb) e;
                         // Paper start
-                        if (!loopItem.dead && !(maxValue > 0 && loopItem.value >= maxValue)) {
+                        if (!loopItem.dead && !(maxValue > 0 && loopItem.value >= maxValue)  && new com.destroystokyo.paper.event.entity.ExperienceOrbMergeEvent((org.bukkit.entity.ExperienceOrb) entity.getBukkitEntity(), (org.bukkit.entity.ExperienceOrb) loopItem.getBukkitEntity()).callEvent()) { // Paper
                             long newTotal = (long)xp.value + (long)loopItem.value;
                             if ((int) newTotal < 0) continue; // Overflow
                             if (maxValue > 0 && newTotal > (long)maxValue) {
-- 
2.21.0

